### PR TITLE
Test Framework - Don't serialize string Json element directly to session

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/core/Azure.Core/tests/RecordSessionTests.cs [hunk 1]
```diff
@@ -84,6 +84,63 @@ public void CanRoundtripSessionRecord(string body, string contentType)
             CollectionAssert.AreEqual(bodyBytes, deserializedRecord.Response.Body);
         }
 
+        [Test]
+        public void CanRoundTripEncodedJson()
```
Comment (christothes) [inline sdk/core/Azure.Core/tests/RecordSessionTests.cs:88]: Would it make sense to either add the same test cases as on `CanRoundtripSessionRecord` or just modify that test to create a `MockRequest` to hold the content as you'd done here?

Comment (pakrym) [reply sdk/core/Azure.Core/tests/RecordSessionTests.cs:88]: Yeah, let's reuse CanRoundtripSessionRecord

**Review by christothes — APPROVED** [review]
👍

**Review by pakrym — CHANGES_REQUESTED** [review]
Semantics of existing tests were changed.

File: sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj [hunk 2]
```diff
@@ -34,6 +34,7 @@
     <Compile Include="..\src\Shared\RetriableStream.cs" />
     <Compile Include="..\src\Shared\RequestRequestContent.cs" />
     <Compile Include="..\src\Shared\ValueStopwatch.cs" />
+    <Compile Include="..\src\Shared\AutoRest\Utf8JsonRequestContent.cs" />
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj]: rem

**Review by pakrym — APPROVED** [review]

**Pull request merged**